controllers/default_controller.py

```diff
@@ -20,15 +19,39 @@ def systemGetAll():
                     {"ciscoXmlName": "NetworkElement", "humanReadableName": "Fabric Interconnects"},
                     {"ciscoXmlName": "EquipmentFex", "humanReadableName": "FEX"},
                     {"ciscoXmlName": "computeRackUnit", "humanReadableName": "Servers"}]
-        finalObjs = []
+        finalObjs = {}
         for x in elements:
             units = []
             components = globalhandle.query_children(in_dn="sys", class_id=x["ciscoXmlName"])
-            componentObj = {
-                "name": x["humanReadableName"],
-                "members": units}
             for y in components:
                 subElement = {"relative_path": "/"+(vars(y))["dn"]}
                 units.append(subElement)
-            finalObjs.append(componentObj)
-        return finalObjs
+            finalObjs[x["humanReadableName"]]=units
+    globalhandle.logout()
+    return finalObjs
+
+def getRackmount():
+    global globalhandle
```

Comment:
Instead of using login to create a global handle, I think you should just pass the username and password to this controller.  That way the controller can login, get the ucs data, and then logout.